ariadne_codegen/client_generators/arguments.py

```diff
@@ -63,6 +63,7 @@ def generate(
                 convert_to_snake_case=self.convert_to_snake_case,
                 plugin_manager=self.plugin_manager,
                 node=variable_definition,
+                trim_leading_underscore=True,
```

Comment:
Imo we shouldn't trim underscores from arguments names, they are not part of pydantic's models so there is no harm in leaving them.

Reply:
I'll of course change if you want to but I thought about it and this felt like the proper way if the type is changed in a class.

So f.ex. for this, I think it could've made sense to keep it:

```graphql
type Mutation {
  add_str(_arg: String!): String!
}

mutation TestStr($_arg: String!) {
  add_str(_arg: $_arg)
}
```

This would've generate this instead of `arg: str`:
```python
async def test_str(self, _arg: str) -> TestStr
```

However, if the return type would be a custom type like this:
```graphql
type T {
  _a: String!
}

input TInput {
  _a: String!
}

type Mutation {
  add_t(_t: TInput!): T!
}

mutation TestT($_t: TInput!) {
  add_t(_t: $_t) {
    _a
  }
}
```

We now generate 
```python
async def test_t(self, _t: TInput) -> TestT
```

And if we unwrap `TestT` we would have 
```python
class TestT(BaseModel):
    add_t: "TestTAddT"


class TestTAddT(BaseModel):
    a: str = Field(alias="_a")
```

Where `_a` that holds the string is now replaced.

I know they're different types (inputs vs. return types) but I think a consistent behaviour of GraphQL fields with leading underscores would make sense.

Reply:
Actually after reading this, I think you're right, they're different things so no need to change this for non pydantic stuff! I'll change this!